airflow-core/src/airflow/exceptions.py

```diff
@@ -558,3 +558,14 @@ def __str__(self):
 
 class UnknownExecutorException(ValueError):
     """Raised when an attempt is made to load an executor which is not configured."""
+
+
+class DeserializationError(Exception):
+    """Raised when a DAG cannot be deserialized."""
+
+    def __init__(self, dag_id: str, original_error: Exception, error_type: str = "general"):
+        self.dag_id = dag_id
+        self.original_error = original_error
+        self.error_type = error_type
```

Comment:
I think you can get rid of the constructor and use exception chaining to raise a `DeserializationError` without losing the underlying exception information:
https://peps.python.org/pep-3134/

Basically `raise DeserializationError(...) from e`